notebooks/icrp107_dataset/icrp107_dataset.ipynb

```diff
@@ -261,17 +262,111 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Prepare a pandas DataFrame for the ICRP-107 decay data."
+    "Create a pandas DataFrame to hold AME2020 atomic data, needed for atomic masses of stable progeny. Pull data from the AME2020 `.txt` file (`mass_1.mas20.txt`)."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 4,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# read and parse AME2020 file for atomic masses\n",
+    "ame2020_format = ff.FortranRecordReader('(a1,a3,a5,a5,a5,1x,a3,a4,1x,a14,a12,a13,1x,a10,1x,a2,a13,a11,1x,a3,1x,a13,a12)')\n",
+    "\n",
+    "with open('mass_1.mas20.txt','r') as ame2020:\n",
+    "    ame2020_lines = ame2020.readlines()\n",
+    "\n",
+    "ame2020_df = pd.DataFrame()\n",
+    "isotope_dictionary_list = {}\n",
+    "\n",
+    "for i in range(len(ame2020_lines)-37):\n",
+    "    # starting from the first isotope on line 38, read each line seperately\n",
+    "    isotope = ame2020_format.read(ame2020_lines[i+37])\n",
+    "    # name is combination of element and number of nucleons\n",
+    "    name = isotope[5].split()[0] + \"-\" + str(isotope[4]).split()[0]\n",
+    "    Z = int(isotope[3].split()[0])\n",
+    "    A = int(isotope[4].split()[0])\n",
+    "    # concatenate atomic number with decimal places, in μ amu\n",
+    "    # split by '#' to remove characters prior to float conversion\n",
+    "    # divide by 10e6, as AME2020 is in micro amu\n",
+    "    mass = (float(isotope[14].strip(\"# \")\n",
+    "           + isotope[15].strip(\"# \")))/10**6\n",
+    "    # build dictionary for specific isotope\n",
+    "    isotope_dictionary_list[name] = [Z, A, mass]\n",
+    "\n",
+    "# convert from dictionaries to DataFrame for max speed\n",
+    "ame2020_df = pd.DataFrame.from_dict(isotope_dictionary_list,\n",
+    "                                        orient='index')\n",
+    "ame2020_df.columns = [\"Z\",\"A\",\"Mass\"] "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create a pandas DataFrame to hold NuBase atomic data, needed for the isomer excitation energy of the metastable states. Pull data from the associated file (`nubase_3.mas20.txt`)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# read and parse AME2020 file for atomic masses\n",
+    "nubase_format = ff.FortranRecordReader('(a4,a7,a5,a2,a13,a11,a12)')\n",
```

Comment:
Well done for working out the Nubase2020 Fortran data format, and solving the challenges such as the float entries that need to read as strings, and the extra spaces that aren't listed in the data format table! Maybe consider adding these extra spaces in separately as extra `a1,`s, as this makes the format string easier to cross check against nubase_3.mas20.txt (see example in the notebook I sent on Slack).